application-analytics-default/src/main/java/com/xwiki/analytics/internal/MostViewedJsonNormaliser.java

```diff
@@ -0,0 +1,200 @@
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.analytics.internal;
+
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Iterator;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.resource.CreateResourceReferenceException;
+import org.xwiki.resource.CreateResourceTypeException;
+import org.xwiki.resource.ResourceReference;
+import org.xwiki.resource.ResourceReferenceResolver;
+import org.xwiki.resource.ResourceType;
+import org.xwiki.resource.ResourceTypeResolver;
+import org.xwiki.resource.UnsupportedResourceReferenceException;
+import org.xwiki.resource.entity.EntityResourceReference;
+import org.xwiki.url.ExtendedURL;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.xwiki.analytics.JsonNormaliser;
+
+import liquibase.repackaged.org.apache.commons.lang3.exception.ExceptionUtils;
+
+/**
+ * The normaliser for the MostViewedMacro.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named("MostViewedPages")
+@Singleton
+public class MostViewedJsonNormaliser implements JsonNormaliser
+{
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private static final String DATE = "date";
+
+    private static final String URL = "url";
+
+    private static final String LABEL = "label";
+
+    @Inject
+    private ResourceReferenceResolver<ExtendedURL> resourceReferenceResolver;
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private ResourceTypeResolver<ExtendedURL> resourceTypeResolver;
+
+    @Inject
+    @Named("compactwiki")
+    private EntityReferenceSerializer<String> serializer;
+
+    /**
+     * This method will normalise the jsons returned by Matomo into a single format.
```

Comment:
`Normalize Matomo response for format consistency and add extra information needed by XWiki.`